src/exchange/account/order/exchange-account-order.controller.js

```diff
@@ -0,0 +1,114 @@
+angular
+  .module('Module.exchange.controllers')
+  .controller(
+    'exchangeAccountOrderCtrl',
+    class ExchangeAccountOrderCtrl {
+      constructor(
+        $q,
+        $scope,
+        $translate,
+        $window,
+
+        Exchange,
+        exchangeOrder,
+        exchangeServiceInfrastructure,
+        messaging,
+        navigation,
+        User,
+      ) {
+        this.$q = $q;
+        this.$scope = $scope;
+        this.$translate = $translate;
+        this.$window = $window;
+
+        this.Exchange = Exchange;
+        this.exchangeOrder = exchangeOrder;
+        this.exchangeServiceInfrastructure = exchangeServiceInfrastructure;
+        this.messaging = messaging;
+        this.navigation = navigation;
+        this.User = User;
+      }
+
+      $onInit() {
+        this.$routerParams = this.Exchange.getParams();
+        this.serviceToTarget = this.navigation.currentActionData.serviceToTarget;
+
+        this.renewalPeriod = 'P1M';
+        this.updateValueOfServiceToTargetExpiresBeforeEndOfRenewalPeriod();
+
+        return this.fetchingOffers();
+      }
+
+      fetchingOffers() {
+        this.isFetchingOffers = true;
+
+        return this.exchangeOrder
+          .fetchingProductOffers()
+          .then((offers) => {
+            this.offers = offers;
+          })
+          .catch((error) => {
+            this.messaging.writeError(
+              this.$translate.instant(
+                'exchange_account_order_fetchingOffers_error',
+                {
+                  errorMessage: _.get(error, 'data.message', error.toString()),
+                },
+              ),
+            );
+
+            this.$scope.resetAction();
+          })
+          .finally(() => {
+            this.isFetchingOffers = false;
+          });
+      }
+
+      updateValueOfServiceToTargetExpiresBeforeEndOfRenewalPeriod() {
```

Comment:
Maybe this could be exposed in a `service` as it is common to several `controllers`

Reply:
I've intentionally duplicated code as there are lots of chances that the 3 dialogs will evolve in different ways